geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/tube/TubeBuilder.scala

```diff
@@ -97,6 +97,55 @@ abstract class TubeBuilder(val tubeFeatures: SimpleFeatureCollection,
     }
   }
 
+  /**
+    * This function checks if a segment crosses the IDL.
+    * @param point1 The first point in the segment
+    * @param point2 The second point in the segment
+    * @return boolean true if the segment crosses the IDL, otherwise false
+    */
+  def crossesIDL(point1:Coordinate,point2:Coordinate): Boolean = {
+    ((point1.x > 0 && point2.x < 0) || (point2.x > 0 && point1.x < 0))
```

Comment:
wouldn't this also catch points along the prime meridian? I believe the standard check done in JTS is if the points are more than 180 degrees apart.

Reply:
yeah we may also want to make sure the unit test tests this with a line directly on it?